src/javascript/app/pages/trade/charts/highchart.js

```diff
@@ -579,7 +579,7 @@ const Highchart = (() => {
     const endContract = () => {
         if (chart && !stop_streaming) {
             drawLineX({
-                value     : (isSoldBeforeExpiry() ? sell_time : end_time),
+                value     : (isSoldBeforeExpiry() ? (sell_time || sell_spot_time) : end_time),
```

Comment:
we need to test this for all types of trades, to make sure it doesn't break anything.